optuna/samplers/_tpe/probability_distributions.py

```diff
@@ -0,0 +1,123 @@
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+import numpy as np
+
+from optuna.samplers._tpe import _truncnorm
+
+
+class _BatchedCategoricalDistributions(NamedTuple):
+    weights: np.ndarray
+
+
+class _BatchedTruncNormDistributions(NamedTuple):
+    mu: np.ndarray
+    sigma: np.ndarray
+    low: float  # Currently, low and high do not change per trial.
+    high: float
+
+
+class _BatchedDiscreteTruncNormDistributions(NamedTuple):
+    mu: np.ndarray
+    sigma: np.ndarray
+    low: float  # Currently, low, high and step do not change per trial.
+    high: float
+    step: float
+
+
+_BatchedDistributions = Union[
+    _BatchedCategoricalDistributions,
+    _BatchedTruncNormDistributions,
+    _BatchedDiscreteTruncNormDistributions,
+]
+
+EPS = 1e-12
```

Comment:
Now that `_normal_cdf` was removed, `EPS` is not used in this file.
```suggestion
```

Reply:
Thanks for pointing it out!
I applied the suggestion.